sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -0,0 +1,278 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitutionLocation,
+  createFakeStudentAppeal,
+  getProviderInstanceForModule,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import {
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import {
+  ApplicationStatus,
+  AssessmentTriggerType,
+  InstitutionLocation,
+  StudentScholasticStandingChangeType,
+} from "@sims/sims-db";
+import {
+  APPLICATION_NOT_FOUND,
+  FormNames,
+  FormService,
+  INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+} from "../../../../services";
+import { APPLICATION_CHANGE_NOT_ELIGIBLE } from "../../../../constants";
+import { ScholasticStandingAPIInDTO } from "../../models/student-scholastic-standings.dto";
+import { getISODateOnlyString } from "@sims/utilities";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+import { TestingModule } from "@nestjs/testing";
+
+describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticStanding.", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let collegeFLocation: InstitutionLocation;
+  let formService: FormService;
+  let payload: ScholasticStandingAPIInDTO;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // Mock the form service to validate the dry-run submission result.
+    // TODO: Form service must be hosted for E2E tests to validate dry run submission
+    // and this mock must be removed.
+    formService = await getProviderInstanceForModule(
+      appModule,
+      AppInstitutionsModule,
+      FormService,
+    );
+  });
+
+  beforeEach(async () => {
+    mockFormioDryRun();
+  });
+
+  it("Should throw bad request exception error when the payload is invalid for formIO dryRun test.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    const invalidPayload = {
+      data: {
+        booksAndSupplies: 1000,
+        scholasticStandingChangeType:
+          StudentScholasticStandingChangeType.ChangeInIntensity,
+      },
+    };
+    mockFormioDryRun({ validDryRun: false, payload: invalidPayload });
```

Comment:
I am ok to proceed but the `invalidPayload` here is not changing anything, right?
The expected error returned from the API will be only based on `validDryRun: false`, which means that `invalidPayload` can be removed, IMO.